pysmt/cmd/installers.py

```diff
@@ -0,0 +1,630 @@
+        self.archive_path = os.path.join(self.base_dir, self.archive_name)
+
+        if self.mirror_link is not None:
+            self.download_links.append("%s/%s" % (self.mirror_link, self.archive_name))
+        self.download_links.append("https://codeload.github.com/CVC4/CVC4/tar.gz/%s"\
+                                    % (self.git_version))
+        self.extract_path = self.archive_path[:-7] # get rid of '.tar.gz'
+        self.bin_path = os.path.join(self.base_dir, "CVC4_bin")
+
+    def move(self):
+        SolverInstaller.mv(os.path.join(self.bin_path, "share/pyshared/CVC4.py"),
+                           self.bindings_dir)
+        SolverInstaller.mv(os.path.join(self.bin_path, "lib/pyshared/_CVC4.so"),
+                                        self.bindings_dir)
+
+    def compile(self):
+        # Patch the distribution to avoid a known problem
+        patch_name = "cvc4_wrapper.patch"
+        plink = "https://raw.githubusercontent.com/pysmt/solvers_patches/master/%s" % patch_name
+        SolverInstaller.do_download(plink, os.path.join(self.extract_path, patch_name))
+
+        # Apply patch
+        SolverInstaller.run("patch -p1 -i %s" % patch_name,
+                            directory=self.extract_path)
+
+        # Prepare the building system
+        SolverInstaller.run("bash autogen.sh", directory=self.extract_path)
+
+        # Build ANTLR
+        SolverInstaller.run("bash get-antlr-3.4",
+                            directory=os.path.join(self.extract_path, "contrib"))
+
+        # Configure and build CVC4
+        config = "./configure --prefix={bin_path} \
+                              --enable-language-bindings=python \
+                              --with-antlr-dir={dir_path}/antlr-3.4 ANTLR={dir_path}/antlr-3.4/bin/antlr3;\
+                  make; \
+                  make install ".format(bin_path=self.bin_path, dir_path=self.extract_path)
+        SolverInstaller.run(config, directory=self.extract_path)
+
+        # Fix the paths of the bindings
+        SolverInstaller.run("cp CVC4.so.3.0.0 _CVC4.so",
+                            directory=os.path.join(self.bin_path, "lib/pyshared"))
+
+    def get_installed_version(self):
+        with TemporaryPath([self.bindings_dir]):
+            try:
+                import CVC4
+                return CVC4.Configuration_getVersionString()
+            except ImportError:
+                if "CVC4" in sys.modules:
+                    del sys.modules["CVC4"]
+                return None
+
+
+class YicesInstaller(SolverInstaller):
+
+    SOLVER = "yices"
+
+    def __init__(self, install_dir, bindings_dir, solver_version, os_name=None,
+                 architecture=None, python_version=None, mirror_link=None):
+        SolverInstaller.__init__(self, install_dir, bindings_dir, solver_version,
+                                 os_name, architecture, python_version, mirror_link)
+
+        assert self.os_name == "linux"
```

Comment:
This check is only done for yices. Is it because of ```pyices```?

In general, I don't know whether we should perform this type of checks.